nativelib/src/main/resources/scala-native/gc/commix/datastructures/GreyPacket.h

```diff
@@ -54,30 +61,31 @@ bool GreyPacket_IsEmpty(GreyPacket *packet);
 void GreyPacket_MoveItems(GreyPacket *src, GreyPacket *dst, int count);
 
 void GreyList_Init(GreyList *list);
-uint32_t GreyList_Size(GreyList *list);
+UInt24 GreyList_Size(GreyList *list);
 void GreyList_Push(GreyList *list, word_t *greyPacketsStart,
                    GreyPacket *packet);
 void GreyList_PushAll(GreyList *list, word_t *greyPacketsStart,
                       GreyPacket *first, uint_fast32_t size);
 GreyPacket *GreyList_Pop(GreyList *list, word_t *greyPacketsStart);
 
-static inline uint32_t GreyPacket_IndexOf(word_t *greyPacketsStart,
-                                          GreyPacket *packet) {
+static inline UInt24 GreyPacket_IndexOf(word_t *greyPacketsStart,
+                                        GreyPacket *packet) {
     assert(packet != NULL);
     assert((void *)packet >= (void *)greyPacketsStart);
-    return (uint32_t)(packet - (GreyPacket *)greyPacketsStart) + 2;
+    return UInt24_fromUInt32((packet - (GreyPacket *)greyPacketsStart) + 2);
 }
 
 static inline GreyPacket *GreyPacket_FromIndex(word_t *greyPacketsStart,
-                                               uint32_t idx) {
-    assert(idx >= 2);
-    return (GreyPacket *)greyPacketsStart + (idx - 2);
+                                               UInt24 idx) {
+    uint32_t idxValue = UInt24_toUInt32(idx);
+    assert(idxValue >= 2);
+    return (GreyPacket *)greyPacketsStart + (idxValue - 2);
 }
 
 static inline uint64_t GreyPacketRef_Empty() {
     GreyPacketRef initial;
     initial.sep.idx = GREYLIST_LAST;
-    initial.sep.size = 0;
+    initial.sep.size = (UInt24){0};
```

Comment:
Is there any reason not to do
```suggestion
    initial.sep.size = UInt24_fromUInt32(0);
```
here?

Reply:
Not really,  only possible possive value from that might be replacing method call with constant literal. However, after all we have optimizers and this method Is probably not called very often. 
Switched to your proposal as it's more readable and less error prone.